demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1526 @@
+                        suspendTimeout = OTA_SUSPEND_TIMEOUT_MS;
+
+                        while( ( ( state = OTA_GetState() ) != OtaAgentStateSuspended ) && ( suspendTimeout > 0U ) )
+                        {
+                            /* Wait for OTA Library state to suspend */
+                            sleep( OTA_EXAMPLE_TASK_DELAY_MS );
+                            suspendTimeout -= OTA_EXAMPLE_TASK_DELAY_MS;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of demo.
+ *
+ * This example initializes the OTA library to enable OTA updates via the
+ * MQTT broker. It simply connects to the MQTT broker with the users
+ * credentials and spins in an indefinite loop to allow MQTT messages to be
+ * forwarded to the OTA agent for possible processing. The OTA agent does all
+ * of the real work; checking to see if the message topic is one destined for
+ * the OTA agent. If not, it is simply ignored.
+ */
+int main( int argc,
+          char ** argv )
+{
+    ( void ) argc;
+    ( void ) argv;
+
+    /* Return error status. */
+    int returnStatus = EXIT_SUCCESS;
+
+    /* Semaphore initialization flag. */
+    bool bufferSemInitialized = false;
+    bool mqttMutexInitialized = false;
+
+    /* Initialize semaphore for buffer operations. */
+    if( sem_init( &bufferSemaphore, 0, 1 ) != 0 )
+    {
+        LogError( ( "Failed to initialize buffer semaphore"
+                    ",errno=%s",
+                    strerror( errno ) ) );
+
+        returnStatus = EXIT_FAILURE;
+    }
+    else
+    {
+        bufferSemInitialized = true;
+    }
+
+    /* Initialize mutex for coreMQTT APIs. */
+    if( pthread_mutex_init( &mqttMutex, NULL ) != 0 )
+    {
+        LogError( ( "Failed to initialize mutex for mqtt apis"
+                    ",errno=%s",
+                    strerror( errno ) ) );
+
+        returnStatus = EXIT_FAILURE;
+    }
+    else
+    {
+        mqttMutexInitialized = true;
+    }
+    
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Initialize MQTT library. Initialization of the MQTT library needs to be
+         * done only once in this demo. */
+        returnStatus = initializeMqtt( &mqttContext, &networkContext );
+    }
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Start OTA demo. */
+        returnStatus = startOTADemo();
+    }
+
+    if( bufferSemInitialized == true )
+    {
+        /* Cleanup semaphore created for buffer operations. */
+        if( sem_destroy( &bufferSemaphore ) != 0 )
+        {
+            LogError( ( "Failed to destroy buffer semaphore"
+                        ",errno=%s",
+                        strerror( errno ) ) );
+
+            returnStatus = EXIT_FAILURE;
+        }
+    }
+
+    if( mqttMutexInitialized == true )
+    {
+        /* Cleanup semaphore created for buffer operations. */
```

Comment:
```suggestion
        /* Cleanup mutex created for MQTT operations. */
```

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/bc13b469297f8bfa2c27c3738322e60cc4bee1d1